apps/query-ui/queryui/queryui.py

```diff
@@ -97,14 +124,17 @@ def show_dag(plan: LogicalPlan):
 
 def run_query(query: str, index: str, plan_only: bool, do_trace: bool, use_cache: bool):
     """Run the given query."""
-    trace_dir = None
+    st.session_state.trace_dir = None
     if do_trace:
-        trace_dir = tempfile.mkdtemp()
-        st.write(f"Writing execution traces to `{trace_dir}`")
+        if not plan_only:
+            st.session_state.trace_dir = DEFAULT_TRACE_DIR
```

Comment:
See comment above, I think you can just continue using tempfile.mkdtemp() for this.